fn_grpc_interface/fn_grpc_interface/components/function_grpc.py

```diff
@@ -0,0 +1,292 @@
+# (c) Copyright IBM Corp. 2010, 2018. All Rights Reserved.
+# -*- coding: utf-8 -*-
+# pragma pylint: disable=unused-argument, no-self-use
+"""Function implementation"""
+
+import logging
+from resilient_circuits import ResilientComponent, function, handler, StatusMessage, FunctionResult, FunctionError
+import sys
+import os
+import json
+import grpc
+import re
+import inspect
+from google.protobuf.json_format import MessageToDict
+
+
+class FunctionComponent(ResilientComponent):
+    """Component that implements generic wrapper for the gRPC client on the Resilient platform"""
+
+    def __init__(self, opts):
+        """constructor provides access to the configuration options"""
+        super(FunctionComponent, self).__init__(opts)
+        self.options = opts.get("fn_grpc", {})
+
+    @handler("reload")
+    def _reload(self, event, opts):
+        """Configuration options have changed, save new values"""
+        self.options = opts.get("fn_grpc", {})
+
+    def _get_interface_file_names(self, base_dir=None, files_dir=None):
+        """
+        component that return the absolute path of all files in the given directory
+        :param base_dir:  interface pb2 files parent directory
+        :param files_dir:  interface pb2 files directory i.e same as package name
+        :return: lists of all pb2 files absolute path inside the given directory path
+        """
+        __dir_path = os.path.join(base_dir, files_dir)
+        return os.listdir(__dir_path)
+
+    def _get_grpc_interface_module(self, interface_files=[], base_dir=None, files_dir=None):
+        """
+        component that imports all the modules in the current workspace and returns the given imported modules objects.
+        :param interface_files: List of all buffer pb2 files absolute path to import in the current work space
+        :param base_dir:  interface pb2 files parent directory.
+        :param files_dir: interface pb2 files directory i.e same as package name.
+        :return: returns imported interface pb2 files module objects on the successful execution, else returns None object.
+        """
+        assert isinstance(interface_files, list), "This Parameter "'"interface_files"'" should be list data type"
+        assert base_dir is not None, "The param base_dir should contain parents parent directory of gRPC interface \
+        files"
+        assert files_dir is not None, "The param files_dir should points to parent directory of gRPC interface files"
+
+        if interface_files:
+            abs_file_path_json = dict()
+            interface_module = []
+
+            # creating files abs path
+            for file in interface_files:
```

Comment:
* Should avoid using file here: https://docs.python.org/2/library/functions.html#file

Reply:
ok